ang/civicase/CustomScrollbar.js

```diff
@@ -0,0 +1,62 @@
+(function (angular, $, _) {
+  var module = angular.module('civicase');
+
+  module.directive('civicaseCustomScrollbar', function ($rootScope, $parse, $timeout) {
+    return {
+      restrict: 'A',
+      link: civicaseCustomScrollbarLink
+    };
+
+    /**
+     * Link function for civicaseCustomScrollbar
+     *
+     * @param {Object} $scope
+     * @param {jQuery} element
+     * @param {Object} attrs
+     */
+    function civicaseCustomScrollbarLink ($scope, element, attrs) {
+      var options = {
+        autoHide: false
+      };
+
+      if (attrs.civicaseCustomScrollbar) {
+        options = $parse(attrs.civicaseCustomScrollbar)();
+      }
+
+      (function init () {
+        initScrollbar(options);
+        initSubscribers();
+      }());
+
+      /**
+       * Initiate scrollbar plugin
+       *
+       * @param {Object} options for the simplebar
+       */
+      function initScrollbar (options) {
+        new SimpleBar(element[0], options);
+      }
+
+      /**
+       * Initiate Subscribers
+       */
+      function initSubscribers () {
+        $rootScope.$on('civicase::custom-scrollbar::recalculate', recalculateSubscriber);
+      }
+
+      /**
+       * Subscriber for 'civicase::custom-scrollbar::recalculate' event
+       *
+       * @param {Event} event
+       * @param {DOMElement} element
+       */
+      function recalculateSubscriber (event, element) {
```

Comment:
why does the directive needs to be told from the directive's consumer on what element the simplebar is init'd on? The directive should know that, given that it's its job

Can it be as simple as keeping the reference to the simplebar when instantiated?
```js
var reference = new SimpleBar(element[0], options);
```

in any case the approach seems wrong, a directive's consumer should be concerned of only sending the "recalculate" signal, the rest should be handled by the directive internally

Reply:
doesn't both the controller and link function have access to `element`, which is the element the directive is applied to?

Reply:
also @swastikpareek does the solution that i provided above work? Can't you just keep the reference to the `SimpleBar` instance and call "recalculate" on that?

Reply:
Yes, it worked by the below mentioned solution.
> Can't you just keep the reference to the SimpleBar instance and call "recalculate" on that?